panic with amtool: invalid memory address or nil pointer dereference

running 

`amtool silence add --verbose '--start=2025-01-23T19:30:00Z' '--duration=90m' '--author=Ops Automation' '--comment=Scheduled silence: backups-vm-trv-prd-trv' '--alertmanager.url=http://alertmanager-operated.monitoring.svc.cluster.local:9093/' 'alertname=WindowsCpuUtilization' 'vmName=vm-trv-prd-trv'`

results into a panic:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x9c1801]

goroutine 1 [running]:
github.com/prometheus/common/promslog.(*AllowedLevel).Set(0x0, {0xee4f9c, 0x5})
/go/pkg/mod/github.com/prometheus/common@v0.61.0/promslog/slog.go:117 +0x21
github.com/prometheus/alertmanager/cli.initMatchersCompat(0xc000005c08?)
/app/cli/root.go:55 +0x66
github.com/alecthomas/kingpin/v2.(*actionMixin).applyActions(...)
/go/pkg/mod/github.com/alecthomas/kingpin/v2@v2.4.0/actions.go:28
github.com/alecthomas/kingpin/v2.(*Application).applyActions(0xc000646700?, 0xc0000d66c0)
/go/pkg/mod/github.com/alecthomas/kingpin/v2@v2.4.0/app.go:562 +0x5a
github.com/alecthomas/kingpin/v2.(*Application).execute(0xc000646700, 0xc0000d66c0, {0xc0000bfac0, 0x2, 0x2})
/go/pkg/mod/github.com/alecthomas/kingpin/v2@v2.4.0/app.go:398 +0x65
github.com/alecthomas/kingpin/v2.(*Application).Parse(0xc000646700, {0xc00013e010?, 0xc00013e010?, 0xc0005617a0?})
/go/pkg/mod/github.com/alecthomas/kingpin/v2@v2.4.0/app.go:230 +0x14a
github.com/prometheus/alertmanager/cli.Execute()
/app/cli/root.go:183 +0x12dc
main.main()
/app/cmd/amtool/main.go:19 +0xf

```

--- a/cli/root.go
+++ b/cli/root.go
@@ -52,6 +52,7 @@ var (
 func initMatchersCompat(_ *kingpin.ParseContext) error {
 	promslogConfig := &promslog.Config{Writer: os.Stdout}
 	if verbose {
+		promslogConfig.Level = &promslog.AllowedLevel{}
 		_ = promslogConfig.Level.Set("debug")
 	}
 	logger := promslog.New(promslogConfig)

